### ISSUE TITLE
Cannot install from secure PyPI instance without manual cred entry; envvars not used

### ISSUE THREAD

**Issue by gregdan3** [pdm-project/pdm#1281]
- [X] I have searched the issue tracker and believe that this is not a duplicate.

## Steps to reproduce

Attempt to `pdm sync` with a custom pypi instance in the `pyproject.toml` such as the following, including environmental variables shown:

```toml
[[tool.pdm.source]]
url = "https://${USERNAME}:${PASSWORD}@my.pypi.instance.com/simple"
verify_ssl = true
name = "pypi"
```

## Actual behavior
`pdm` requests a username and password for the pypi instance instead of reading the environment, failing on any packages exclusively sourced from it if `-v` is not passed on `pdm sync`.

Example from Docker:
```
#0 49.76 pdm.termui: Error occurs: 
#0 49.76 Traceback (most recent call last):
#0 49.76   File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
#0 49.76     result = self.fn(*self.args, **self.kwargs)
#0 49.76   File "/usr/local/lib/python3.10/site-packages/pdm/installers/synchronizers.py", line 217, in install_candidate
#0 49.76     self.manager.install(can)
#0 49.76   File "/usr/local/lib/python3.10/site-packages/pdm/installers/manager.py", line 39, in install
#0 49.76     installer(str(prepared.build()), self.environment, prepared.direct_url())
#0 49.76   File "/usr/local/lib/python3.10/site-packages/pdm/models/candidates.py", line 337, in build
#0 49.76     self.obtain(allow_all=False)
#0 49.76   File "/usr/local/lib/python3.10/site-packages/pdm/models/candidates.py", line 405, in obtain
#0 49.76     result = finder.download_and_unpack(
#0 49.76   File "/usr/local/lib/python3.10/site-packages/unearth/finder.py", line 356, in download_and_unpack
#0 49.76     file = unpack_link(
#0 49.76   File "/usr/local/lib/python3.10/site-packages/unearth/preparer.py", line 300, in unpack_link
#0 49.76     resp = session.get(link.normalized, stream=True)
#0 49.76   File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 600, in get
#0 49.76     return self.request("GET", url, **kwargs)
#0 49.76   File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 587, in request
#0 49.76     resp = self.send(prep, **send_kwargs)
#0 49.76   File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 708, in send
#0 49.76     r = dispatch_hook("response", hooks, r, **kwargs)
#0 49.76   File "/usr/local/lib/python3.10/site-packages/requests/hooks.py", line 30, in dispatch_hook
#0 49.76     _hook_data = hook(hook_data, **kwargs)
#0 49.76   File "/usr/local/lib/python3.10/site-packages/unearth/auth.py", line 226, in handle_401
#0 49.76     username, password, save = self._prompt_for_password(parsed.netloc)
#0 49.76   File "/usr/local/lib/python3.10/site-packages/pdm/models/auth.py", line 38, in _prompt_for_password
#0 49.76     return super()._prompt_for_password(netloc)
#0 49.76   File "/usr/local/lib/python3.10/site-packages/unearth/auth.py", line 189, in _prompt_for_password
#0 49.76     username = input(f"User for {netloc}: ")
#0 49.76 EOFError: EOF when reading a line
```

## Expected behavior
`pdm` should examine the environment and pull the corresponding variables, thus not prompting for a username/password for your secured pypi instance. It was possible in `pdm~=1.15` and many earlier versions to fetch these credentials from the environment.

## Environment Information

```bash
PDM version:
  2.0.3
Python Interpreter:
  /usr/bin/python3.10 (3.10)
Project Root:
  /home/gregdan3/git/temp
Project Packages:
  /home/gregdan3/git/temp/__pypackages__/3.10
{
  "implementation_name": "cpython",
  "implementation_version": "3.10.5",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "5.18.12-arch1-1",
  "platform_system": "Linux",
  "platform_version": "#1 SMP PREEMPT_DYNAMIC Fri, 15 Jul 2022 15:33:02 +0000",
  "python_full_version": "3.10.5",
  "platform_python_implementation": "CPython",
  "python_version": "3.10",
  "sys_platform": "linux"
}
```

**Comment by frostming** [comment]
What is the `url` in the `pdm.lock` file? Do they start with `https://my.pypi.instance.com/simple`?

**Comment by frostming** [comment]
It is fixed in `unearth` 0.6.1, please update its version.

**Comment by gregdan3** [comment]
This worked. Thanks!
